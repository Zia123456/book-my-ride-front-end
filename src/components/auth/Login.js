/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate, Link } from 'react-router-dom';
import { userLogin } from '../../redux/auth/authActions';

function Login() {
  const { loading, error, isAuthenticated } = useSelector(
    (state) => state.auth,
  );
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const dispatch = useDispatch();

  const navigate = useNavigate();

  const handleLogin = (e) => {
    e.preventDefault();
    const user = { email, password };
    dispatch(userLogin(user));
  };

  // eslint-disable-next-line consistent-return
  useEffect(() => {
    if (isAuthenticated) {
      navigate('/');
    }
  }, [navigate, isAuthenticated]);

  return (
    <div className="loginPage">
      <form onSubmit={handleLogin} className="form">
        <div className="top">
          <span className="color-white">
            Don&apos;t have an account yet?&nbsp;
            <Link to="/register">Register Here</Link>
          </span>
          <header>Login</header>
        </div>
        {error && <p className="errorMessage">{error}</p>}
        <div className="input-field">
          <input
            type="email"
            value={email}
            id="email"
            placeholder="email"
            name="email"
            onChange={(e) => setEmail(e.target.value)}
          />
        </div>
        <div className="input-field">
          <input
            type="password"
            value={password}
            placeholder="password"
            onChange={(e) => setPassword(e.target.value)}
          />
        </div>
        <button type="submit" className="submit" disabled={loading}>
          {loading ? 'Spinner' : 'Login'}
        </button>
      </form>
    </div>
  );
}

export default Login;
