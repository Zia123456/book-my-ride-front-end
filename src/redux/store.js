import { configureStore } from '@reduxjs/toolkit';
import authReducer from './auth/authSlice';
import dataReducer from './reservation/reservationsListSlice';
import carReducer from './car/carSlice';
import addReservationReducer from './reservation/addReservationSlice';

const store = configureStore({
  reducer: {
    data: dataReducer,
    auth: authReducer,
    cars: carReducer,
    addReservation: addReservationReducer,
  },
});

export default store;
