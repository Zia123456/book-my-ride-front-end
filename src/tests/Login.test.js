import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import Login from '../components/auth/Login';

it('Login component should render correctly', () => {
  const result = render(
    <React.StrictMode>
      <Router>
        <Provider store={store}>
          <Login />
        </Provider>
      </Router>
    </React.StrictMode>,
  );
  expect(result).toMatchSnapshot();
});
