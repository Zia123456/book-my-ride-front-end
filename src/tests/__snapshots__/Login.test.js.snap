// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="loginPage"
      >
        <form
          class="form"
        >
          <div
            class="top"
          >
            <span
              class="color-white"
            >
              Don't have an account yet? 
              <a
                href="/register"
              >
                Register Here
              </a>
            </span>
            <header>
              Login
            </header>
          </div>
          <div
            class="input-field"
          >
            <input
              id="email"
              name="email"
              placeholder="email"
              type="email"
              value=""
            />
          </div>
          <div
            class="input-field"
          >
            <input
              placeholder="password"
              type="password"
              value=""
            />
          </div>
          <button
            class="submit"
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="loginPage"
    >
      <form
        class="form"
      >
        <div
          class="top"
        >
          <span
            class="color-white"
          >
            Don't have an account yet? 
            <a
              href="/register"
            >
              Register Here
            </a>
          </span>
          <header>
            Login
          </header>
        </div>
        <div
          class="input-field"
        >
          <input
            id="email"
            name="email"
            placeholder="email"
            type="email"
            value=""
          />
        </div>
        <div
          class="input-field"
        >
          <input
            placeholder="password"
            type="password"
            value=""
          />
        </div>
        <button
          class="submit"
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
